docs/technical-design/file-upload.md

```diff
@@ -0,0 +1,93 @@
+# How FileUpload works on MC-Review
```

Comment:
@kelseylistrom I would like the introduction and FileUpload 101 sections to be pretty clear and readable across the team. Could you look at this from a design perspective let me know if anything key is missing.

Also as I wrote in the PR description I wasn't sure if I should move or at least copy/paste some of the accessibility discussion of FileUpload from the doc in Confluence to here as well. Any thoughts on that? Does it feel like you want an accessibility section in this doc? I definitely will link this doc from Confluence at least when I finish